plutus-core/untyped-plutus-core/src/UntypedPlutusCore/Evaluation/Machine/Cek/Debug/Driver.hs

```diff
@@ -0,0 +1,163 @@
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE ConstraintKinds #-}
+{-# LANGUAGE BangPatterns #-}
+{-# LANGUAGE ViewPatterns #-}
+{-# LANGUAGE GADTs #-}
+module UntypedPlutusCore.Evaluation.Machine.Cek.Debug.Driver
+    ( Breakpoint (..)
+    , Breakpoints
+    , DriverState
+    , Cmd (..)
+    , DTerm
+    , runDriver
+    , handleStep
+    , DebugF (..)
+    -- | Reexport some free functions for convenience
+    , F.MonadFree
+    , F.iterM
+    , F.iterTM
+    , F.partialIterT
+    , F.cutoff
+    ) where
+
+import Annotation
+import UntypedPlutusCore
+import UntypedPlutusCore.Evaluation.Machine.Cek.Debug.Internal
+
+import Data.RandomAccessList.Class qualified as Env
+import Data.Word64Array.Word8 hiding (Index, toList)
+import Control.Monad.Reader
+import Control.Monad.Trans.Free as F
+import Control.Lens hiding (Context)
+import Prettyprinter
+import Data.Ix
+import Data.Set
+import Data.Function
+
+-- | Usually a breakpoint is just a line, but let's be  more generally as a srcspan
+newtype Breakpoint = Breakpoint { unBreakpoint :: SrcSpan }
+    deriving newtype (Show, Pretty, Read, Eq, Ord)
+
+-- | Treat them as `Set`s like `SrcSpan`
+type Breakpoints = Set Breakpoint
+
+-- | The `Term`s that makes sense to debug
+type DTerm uni fun = Term NamedDeBruijn uni fun SrcSpans
```

Comment:
How about:

```haskell
data DebuggerSrcSpans = DebuggerSrcSpans
  { _dssUplcPos :: Maybe PLC.SourcePos
  , _dssSourceSpans :: SrcSpans
  }

type DTerm uni fun = Term NamedDeBruijn uni fun DebuggerSrcSpans
```

Also there are many occurrences of `Term NamedDeBruijn uni fun SrcSpans` in `Internal.hs` which should be replaced by `DTerm uni fun`.

Reply:
This I will do in another PR, perhaps also by sharing with original CEK